compiler/next/include/chpl/uast/BytesLiteral.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_BYTESLITERAL_H
+#define CHPL_UAST_BYTESLITERAL_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/uast/Literal.h"
+#include "chpl/uast/StringLiteral.h"
+
+namespace chpl {
+namespace uast {
+
+
+/**
+  This class represents a bytes literal, for example `b"hello"`
+  and `b''' bytes contents here '''`.
+ */
+class BytesLiteral final : public Literal {
+ private:
+  std::string value_;
```

Comment:
Yeah. I don't know of a reason why pointer comparison for comparing string literals is important. We do have UniqueString which is an interned string type and supports pointer comparisons for identifiers.

Reply:
You’re right, identifiers is what it would be more useful. I imagine we end up doing a lot of string comparisons during name resolution. Interning string literals may be useful if we have a bottleneck in constant and we were compiling a language that relied heavily on the use of string literals (for example if we had a lot of functions that took some sort of modes string of took in key\value pairs using string literals). I don't think Chapel is like that though.